packages/@aws-cdk/aws-ecr/lib/public-repository.ts

```diff
@@ -0,0 +1,428 @@
+    if (!isPatternMatch) {
+      errors.push('Repository name must follow the specified pattern: (?:[a-z0-9]+(?:[._-][a-z0-9]+)*/)*[a-z0-9]+(?:[._-][a-z0-9]+)*');
+    }
+
+    if (errors.length > 0) {
+      throw new Error(`Invalid ECR repository name (value: ${repositoryName})${EOL}${errors.join(EOL)}`);
+    }
+  }
+
+  public readonly publicRepositoryName: string;
+  public readonly publicRepositoryArn: string;
+
+  private resourcePolicy?: iam.PolicyDocument;
+
+  constructor(scope: Construct, id: string, props: PublicRepositoryProps = {}) {
+    super(scope, id, {
+      physicalName: props.publicRepositoryName,
+    });
+    if (Stack.of(this).region !== PublicRepositoryBase.PUBLIC_REPO_REGION) {
+      throw new Error(`ECR Public Repository can only be created in ${PublicRepositoryBase.PUBLIC_REPO_REGION}.`);
+    }
+
+    PublicRepository.validateRepositoryName(this.physicalName);
+
+    if (props.resourcePolicy !== undefined) {
+      this.resourcePolicy = new iam.PolicyDocument({ statements: props.resourcePolicy });
+    }
+
+    const repositoryCatalogData = this.renderRepositoryCatalogData(props);
+
+    const resource = new CfnPublicRepository(this, 'Resource', {
+      repositoryName: this.physicalName,
+      // It says "Text", but they actually mean "Object".
+      repositoryPolicyText: Lazy.any({ produce: () => this.resourcePolicy }),
+      repositoryCatalogData,
+    });
+
+    resource.applyRemovalPolicy(props.removalPolicy);
+
+    this.publicRepositoryName = this.getResourceNameAttribute(resource.ref);
+    this.publicRepositoryArn = this.getResourceArnAttribute(resource.attrArn, {
+      region: '',
+      service: 'ecr-public',
+      resource: 'repository',
+      resourceName: this.physicalName,
+    });
+  }
+
+  private renderRepositoryCatalogData(props: PublicRepositoryProps): CfnPublicRepositoryProps['repositoryCatalogData'] | undefined {
+    if (!props.about && !props.description && !props.usage && !props.architectures && !props.operatingSystems) {
+      return undefined;
+    }
+
+    return {
+      RepositoryDescription: props.description,
+      AboutText: props.about,
+      UsageText: props.usage,
+      Architectures: props.architectures?.map(element => element.value),
+      OperatingSystems: props.operatingSystems?.map(element => element.value),
+    };
+  }
+
+  /**
+   * Add a policy statement to the repository's resource policy
+   */
+  public addToResourcePolicy(statement: iam.PolicyStatement): iam.AddToResourcePolicyResult {
+    if (this.resourcePolicy === undefined) {
+      this.resourcePolicy = new iam.PolicyDocument();
+    }
+    this.resourcePolicy.addStatements(statement);
+    return { statementAdded: true, policyDependable: this.resourcePolicy };
+  }
+
+  protected validate(): string[] {
+    const errors = super.validate();
+    errors.push(...this.resourcePolicy?.validateForResourcePolicy() || []);
+    return errors;
+  }
+}
```

Comment:
This is all the same as `Repository` as well.

Reply:
Same as above.